### PR TITLE
Run Appveyor tests on Python 3.7.

### PR DESCRIPTION
**Pull request by Julian-O**
Addresses one of the requirements for #287.

* Miniconda doesn't have a Python 3.7 version yet.
* PyQt5 5.11.2 was improperly built for Python 3.5. Looks like they fixed the problem without incrementing the version number.
* Minor spelling and spacing improvements.

### CODE REVIEW

**Comment by MatthieuDartiailh** [comment]
I must say I am considering re-writing the appveyor config to sinmly use the latest miniconda and then create a dedicated environment to run the tests. It makes easier to support 3.7 and we avoid having to deal with old distributions. WDYT ?

**Comment by Julian-O** [comment]
It depends what your goals are.

The Travis file mainly relies on Miniconda, which is a great way to quickly get a stable environment and actually unit-test the Python.

But one of the motivating factors of 
  * re-writing the installation documentation,
  * producing the Appveyor testing, and 
  * relying as much as possible on PIP in the appveyor testing was the initial difficulty I had installing Enaml on Windows.

So, as much as anything, I wanted the Appveyor script to test that Enaml and its dependencies could be *installed*. 

I think you should test the platforms you claim to (fully) support. Saying you no longer fully support Python 3.5 and 3.6 is an option for the maintainers. It is a question of balancing effort versus target audience.

----

Some related questions:

* Is it time to stop fully supporting PyQt4? Then you could simply install all the dependencies straight from running setup.py.
* Should CI testing include a "generate examples and documentation" build?
* If so, should there be a target in setup.py for documentation dependencies?

**Comment by MatthieuDartiailh** [comment]
We still support 2.7 (at least for another year) so we cannot completely drop pyqt4. Dropping 3.4 will become reasonable once we consolidate Python 3.7 support (at the time being atom is still broken on 3.7).
Typically those platform are harder to test without miniconda but I am fine with testing (3.5 if possible) 3.6 and 3.7 using only pip (I believe this is actually a good idea).
I usually keep an eye on ReadTheDocs for the documentation, but adding one build for docs may make sense. I do not think we need to build the examples since those are tested but now that vtk is on conda and works we should add it.
Given that appveyor is slow, I suggest we focus on the key tests (not all examples) for Appveyor, and run using conda on 2.7 and 3.4 since it will be easier and running fully using pip on the 3.5, 3.6 and 3.7. I will try to take care of the doc build and vtk on Travis.

**Comment by Julian-O** [comment]
Re: PyQt4: That makes complete sense.

Re: Miniconda versus PIP: That makes sense too. I'd merge this change now, and make it a separate (perhaps low priority?) plan.

Re: Docs: I am thinking mainly about feedback/review so you can see that changes (whether to .rst files, example .enaml files, or others) have been properly reflected in the generated comments. Less concerned about automated generation for ReadTheDocs, although I guess it is a possibility. Again, I see this as icing on the cake, not a priority.

Re: Travis versus AppVeyor: I hadn't noticed the speed difference, although I bet I will now it has been pointed out! Doing docs and vtk on Travis only makes sense.

In summary, I agree with everything you've said. I recommend approving this change as a step towards Py3.7 support, and making the other items low-priority future plans.
